### PR TITLE
The pair of butterfly swords are now just a butterfly sword. Singular.

### PR DESCRIPTION
**Pull request by Daved27hundred**
#### Summary
Bugfixes "Rewords the 'pair of butterfly swords' into a single sword."

#### Purpose of change

CDDA doesn't have dual wielding as a mechanic, a memo which seems to have been missed by this weapon for years. It seems that someone at least tried to make it work, making it twice as heavy and require two hands to use to reflect the fact it's supposed to be two blades....It's still kind of hacky and the facade falls apart when you stick your pair of swords into a sheath that can only hold one.
It's been bugging me for years so I finally did something about it. Maybe in some future of the game we'll finally get dual wielding, the butterfly swords will get justice. For now it's just a machete

#### Describe the solution

Cut the weight and volume in half, renamed the sword to just "butterfly sword" and changed the description. Reworked the crafting recipe. (It costed twice as much resources to craft since you were making two swords) I didn't bother changing the ID to singular because that seemed like too much work.

#### Describe alternatives you've considered

At one point I considered leaving the crafting recipe alone but making it make 2 swords. This is stupid. I'll keep this idea in my pocket though.

### CODE REVIEW

**Comment by IdleSol** [comment]
In the form of an alternative.
You could make a recipe for a new scabbard. Something like a scabbard for paired swords on the back. Add a flag paired weapon and use it as a restraint in the scabbard.

File: data/json/items/melee/swords_and_blades.json [hunk 1]
```diff
@@ -2572,17 +2572,16 @@
     "type": "TOOL",
     "symbol": "/",
     "color": "light_gray",
-    "name": { "str": "pair of butterfly swords", "str_pl": "pairs of butterfly swords" },
-    "description": "A matched pair of traditional Shaolin butterfly swords.  They are about the same size as machetes, but have hand guards and wider blades.",
-    "//": "basically 2x the weight, 1.3x the damage, 2.5x the resources of a single machete to cover the fact that it's two weapons, each with a hand guard",
+    "name": { "str": "butterfly sword", "str_pl": "butterfly swords" },
```
Comment (Karol1223) [inline data/json/items/melee/swords_and_blades.json:2575]:
```suggestion
    "name": { "str": "butterfly sword" },
```

This is unnecessary. The ``s`` is added at the end automatically.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Daved27hundred** [comment]
> In the form of an alternative. You could make a recipe for a new scabbard. Something like a scabbard for paired swords on the back. Add a flag paired weapon and use it as a restraint in the scabbard.

Not a good idea. It ignores the problem I brought up with the very concept of a dual-wielded weapon.
even then, how would I code that? It'd be hard to make a holster that can take either two blades or one pair of swords.

**Comment by Skully-Bojangles** [comment]
The vast majority of butterfly swords are intended to be paired and are crafted to nest flat against each other in a single scabbard. If anything, the fact that they're cinquedea-wide down their entire length is more of an issue for scabbards than their combined thickness. Once you remove those gimmicks and the associated jank you're basically left with a stubby dao.

**Comment by Zireael07** [comment]
@Skully-Bojangles In which case the item should be removed, as there is no dual-wielding in CDDA. Period

**Comment by IdleSol** [comment]
That's a strange remark.

What about bows, crossbows, rifles, and two-handed weapons then? A single object held with two hands. What's the fundamental difference between that and a pair of swords? You're not confused about pairs of boots and gloves, are you?

**Comment by Zireael07** [comment]
The difference is that wielding two swords, if it were implemented, should let you deal double the damage of wielding one, and come with a possibility of being disarmed of one of the two, or one of the two breaking. Bows/crossbows/guns/two handed swords etc. do not have this problem as they are one item, not two

**Comment by IdleSol** [comment]
>It'd be hard to make a holster that can take either two blades or one pair of swords

```
  {
    "id": "SHEATH_PAIR_SWORD",
    "type": "json_flag",
    "restriction": "Item must fit in a scabbard",
    "info": "This item can <info>fit in a scabbard</info> of the appropriate size."
  },


  {
    "id": "Butterfly_sword_scabbard",
    "type": "ARMOR",
    "name": { "str": "Butterfly sword scabbard" },
    "description": "test.",
    "weight": "1360 g",
    "volume": "1750 ml",
    "price": "60 USD",
    "price_postapoc": "5 USD",
    "material": [ "leather", "wood" ],
    "symbol": "[",
    "looks_like": "holster",
    "color": "brown",
    "sided": false,
    "material_thickness": 1,
    "pocket_data": [
      {
        "magazine_well": "1250 ml",
        "max_contains_volume": "2 L",
        "max_contains_weight": "4 kg",
        "max_item_length": "100 cm",
        "holster": true,
        "moves": 30,
        "flag_restriction": [ "SHEATH_PAIR_SWORD" ]
      }
    ],
    "use_action": { "type": "holster", "holster_prompt": "Sheath sword", "holster_msg": "You sheath your %s" },
    "flags": [ "BELTED", "OVERSIZE", "WATER_FRIENDLY" ],
    "armor": [
      {
        "encumbrance": [ 3, 4 ],
        "coverage": 15,
        "covers": [ "leg_l", "leg_r" ],
        "specifically_covers": [ "leg_hip_r", "leg_hip_l" ]
      }
    ],
    "melee_damage": { "bash": 4 }
  },

  {
    "id": "pair_butterfly_swords",
    "type": "TOOL",
    "symbol": "/",
    "color": "light_gray",
    "name": { "str": "pair of butterfly swords", "str_pl": "pairs of butterfly swords" },
    "description": "A matched pair of traditional Shaolin butterfly swords.  They are about the same size as machetes, but have hand guards and wider blades.",
    "//": "basically 2x the weight, 1.3x the damage, 2.5x the resources of a single machete to cover the fact that it's two weapons, each with a hand guard",
    "price": "500 USD",
    "price_postapoc": "60 USD",
    "material": [ "steel" ],
    "flags": [ "DURABLE_MELEE", "SHEATH_PAIR_SWORD", "ALWAYS_TWOHAND" ],
    "weapon_category": [ "SHORT_SWORDS" ],
    "techniques": [ "WBLOCK_2" ],
    "weight": "1766 g",
    "volume": "1500 ml",
    "longest_side": "50 cm",
    "to_hit": { "grip": "weapon", "length": "short", "surface": "line", "balance": "good" },
    "category": "weapons",
    "qualities": [ [ "CUT", 1 ], [ "BUTCHER", 8 ] ],
    "melee_damage": { "bash": 6, "cut": 30 }
  },
```
One item: scabbard for a pair of swords
One item: a pair of swords

A pair of swords is stowed in a scabbard. For authenticity, these scabbards take up both legs.

Perhaps I should clarify. I was going on the premise that you like using these swords. But you don't like having a pair of swords tucked away in a scabbard designed for a single sword. So I suggested adding special scabbards instead of removing these swords from the game. I don't stand by that decision.

>The difference is that wielding two swords, if it were implemented, should let you deal double the damage of wielding one, and come with a possibility of being disarmed of one of the two, or one of the two breaking. Bows/crossbows/guns/two handed swords etc. do not have this problem as they are one item, not two

You're right about something. Then let me draw your attention to brass knuckles. More specifically, the entire melee weapons tab. And I think there were some more paired weapons, but I can't remember.

On the comments themselves. I have no knowledge of weapons, so I may be wrong. Is the hit rate dependent on the weight of the weapon? Or is there a way to change it via a json entry?

Theoretical method of solution:
1. Adding a technique that increases the strike speed by 2 times. Simulating the alternate strike of two swords.
2. Adding a limitation on the technique: it cannot be used when you are grabbed or disarmed. Simulation of losing one weapon or not being able to use it if you are grabbed by hands.
3.Theoretically, you could address “non_functional”: “single sword”. But this goes against the existing repair system. So I don't know the possibility to realize the breakage of one sword, without damaging the second sword.

I haven't tested these ways. So I'm not sure if it's possible to implement, without changing the existing functions.

And in order of crazy ideas: use charges. The simplest example: rocks. You can take two rocks at the same time. You can hit with them, you can throw one of them and stay with the second one.

**Comment by Zireael07** [comment]
> unarmed weapons tab. And I think there were some more paired weapons, but I can't remember.

Wow, TIL that those are dual/paired too. In which case I withdraw my objections.

**Comment by Skully-Bojangles** [comment]
Yeah, but how do you handle hand dominance? Having an arm grappled, disarmed or broken? I'm all for dual wielding and shaolin weaponry, but for the time being, I don't think it's feasible to try and jury-rig a bunch of systems to simulate that when the attack vector/limb rework is still ongoing. The unarmed weapons function under the assumption that you're just strapping them to your hands and fighting unarmed as you would normally, but nobody boxes one-handed the same way one uses a saber. Alternatively, you don't need that much training to avoid getting your right fist tangled on your left.

Plus, once you get handedness down, proper dual wielding is a whole other bag of worms. Dual wielding shouldn't just provide a 1/1 damage multiplier in the same way that people don't dual wield fists by simultaneously punching with both hands. Properly dual wielding without injuring oneself takes considerable coordination and practice, and even then the majority of two weapon fighting styles are asymmetrical and many the offhand to facilitate parrying more than anything else.

I think figuring out some system that allows one to wield stuff in the offhand is important, eventually, especially for stuff like shields or carry luggage, but for now having a handful of items that simulate paired weapons feels bad when you can't also grab two chef knives and do the same thing.

**Comment by Maddremor** [comment]
We simply do not have the infrastructure to support dual-wielding right now, and hacks like these swords have been soundly rejected before. Their presence in the game right now is an oversight.

**Comment by Daved27hundred** [comment]
> We simply do not have the infrastructure to support dual-wielding right now, and hacks like these swords have been soundly rejected before. Their presence in the game right now is an oversight.

thank you for saying it for me. I do appreciate your insight @IdleSol but this is going to have to do until we have an actual framework for dual wielding.

**Pull request merged**